wallet/src/Pos/Wallet/Web/Pending/Submission.hs

```diff
@@ -166,9 +174,63 @@ submitAndSavePtx submitTx PtxSubmissionHandlers{..} ptx@PendingTx{..} = do
         -- while transaction creation failed, due to protocol error or bug,
         -- then we better not remove this pending transaction
         void $ usingPtxCoords removeOnlyCreatingPtx ptx
+
     badInitPtxCondition = InternalError "Expected PtxCreating as initial pending condition"
 
     reportSubmitted ack =
         logDebug $
         sformat ("submitAndSavePtx: transaction submitted with confirmation?: "
                 %build) ack
+
+
+-- As simple and as clear as I could.
+submitAndSavePtxStates
+    :: TxSubmissionMode ctx m
+    => PendingTx
+    -> Timestamp
+    -> ((TxId, TxAux) -> m TxSubmissionResult)
+    -> m TxSubmissionResult
+submitAndSavePtxStates ptx@PendingTx{..} now mSaveTx =
+
+    case _ptxCond of
+
+      PtxApplying poolInfo -> do
+          let Just creationTime = poolInfo ^. thTimestamp
+          if diffTimestamp now creationTime > hour 1
+              then pure $ TxTimeoutWhenApplying poolInfo ptx
+              else pure $ TxStillApplying
+
+      _ -> do
+          addOnlyNewPendingTx ptx
+          (mSaveTx (_ptxTxId, _ptxTxAux) `catches` handlers) `onException` creationFailedHandler
+  where
+    handlers =
+        [ Handler $ \e ->
+            if isReclaimableFailure e
+                then minorError "reclaimable" (SomeException e)
+                else nonReclaimableError (SomeException e)
+
+        , Handler $ \e@SomeException{} ->
+            -- I don't know where this error can came from,
+            -- but it's better to try with tx again than to regret, right?
+            minorError "unknown error" e
+        ]
+
+    minorError desc e = pure $ TxMinorError desc e
+    nonReclaimableError e = pure $ TxNonReclaimableError e
+    creationFailedHandler = pure $ TxCreationFailed
+
+
+submitAndSaveTxApplying
```

Comment:
This function semantics doesn't correspond to its name )

Reply:
Propose a better name, then, and let's get this one over with.
I have no intention of changing the name 5 times guessing what you had in mind.

Reply:
It does not submit or save anything.
How about `cancelPtxWithTimeout`?